shared/src/test/scala/FetchTests.scala

```diff
@@ -638,6 +638,44 @@ class FetchTests extends FreeSpec with Matchers {
 
     totalFetched(rounds) shouldEqual 0
   }
+
+  case class ForgetfulCache() extends DataSourceCache {
```

Comment:
Compiler throws a warning when marking inner case classes as final, i'll leave it for now.